src/main/java/seedu/address/model/modelTeacher/Teacher.java

```diff
@@ -16,16 +17,75 @@
  * values are validated, immutable.
  */
 public class Teacher extends Person {
-
   private final Salary salary;
+  private Set<ID> assignedCoursesID = new HashSet<>();
+  private String assignedCoursesWithNames;
+  private final ID id;
 
   /**
    * Every field must be present and not null.
    */
-  public Teacher(Name name, Phone phone, Email email, Salary salary, Address address,
+  public Teacher(Name name, ID id, Phone phone, Email email, Salary salary, Address address,
+      Set<Tag> tags) {
+    super(name, phone, email, address, tags);
+    this.id = id;
+    this.assignedCoursesWithNames = "None";
+    this.salary = salary;
+    this.tags.addAll(tags);
+  }
+
+  public Teacher(Name name, ID id, Phone phone, Email email, Salary salary, Address address, Set<ID> assignedCoursesID,
       Set<Tag> tags) {
     super(name, phone, email, address, tags);
+    this.id = id;
+    this.assignedCoursesWithNames = "None";
     this.salary = salary;
+    this.assignedCoursesID.addAll(assignedCoursesID);
+    this.tags.addAll(tags);
+  }
+
+  /**
+   * Get id of the teacher.
+   */
+  public ID getID() {
+    return id;
+  }
+
+  /**
+   * Returns an immutable ID set, which throws {@code UnsupportedOperationException} if
+   * modification is attempted.
+   */
+  public Set<ID> getAssignedCoursesID() {
+    return Collections.unmodifiableSet(assignedCoursesID);
+  }
+
+  public String getAssignedCoursesWithNames(){
+    return this.assignedCoursesWithNames;
+  }
+  /**
+   * Converts internal list of assigned student IDs into the name with the IDs
+   */
+  public void processAssignedCourses(FilteredList<Course> filteredCourses){
```

Comment:
what is this method and why is it under teacher? when it deals with assigned student IDS

Reply:
Sorry the comment is wrong, it basically converts the list of course id in the teacher into (course name, id) tuple